src/glfw/window.cc

```diff
@@ -179,65 +179,62 @@ Window::setFullscreen() const
         getFramebufferHeight(),
         GLFW_DONT_CARE );
 
-    logGLFWaction( "choose fullscreen mode", getHandleAddressString( this ) );
+    logGLFWaction( "Set fullscreen mode", this );
 } // setFullscreen
 
-vk::SurfaceKHR
-Window::createWindowSurface(
-    vk::Instance instance,
-    Window& window //
-)
+UniqueSurface
+Window::createSurface( const vk::Instance& instance ) const
 {
-    assert( instance != nullptr );
+    assert( instance );
 
-    vk::SurfaceKHR surface{};
+    VkSurfaceKHR surface;
     glfwCreateWindowSurface(
         static_cast<VkInstance>( instance ),
-        window.get(),
+        m_handle.get(),
         nullptr,
-        reinterpret_cast<VkSurfaceKHR*>( &surface ) //
+        &surface //
     );
 
-    return surface;
+    return vk::UniqueSurfaceKHR{
+        surface,
+        instance }; // Very crucial to pass instance for the correct deleter with dynamic dispatch to work. This is the
+                    // same reason why all files need to be compiled together.
 } // createWindowSurface
 
 WindowManager::WindowManager( ErrorCallbackSignature* error_callback )
 {
     glfwSetErrorCallback( error_callback );
     glfwInit();
-
-    logGLFWaction( "initialize library" );
+    logGLFWaction( "Initialize library" );
 } // WindowManager
 
-WindowManager::~WindowManager()
+WindowManager::~WindowManager() noexcept( false )
 {
     glfwTerminate();
-
-    logGLFWaction( "terminate library" );
+    logGLFWaction( "Terminate library" );
 } // ~WindowManager
 
 bool
 WindowManager::isSuitableVersion()
 {
-    WindowAPIversion version{};
-
+    WindowAPIversion version;
     glfwGetVersion( &version.major, &version.minor, nullptr );
-
     return version >= WindowManager::k_api_min_version;
 } // isSuitableVersion
 
 void
 WindowManager::initialize()
 {
     static WindowManager instance{ errorCallback };
-
-    if ( !isSuitableVersion() )
+    if ( isSuitableVersion() )
     {
-        throw Error{
-            Error::k_user_error,
-            "GLFW minimal version: " + getMinVersionString() + "." //
-        };
+        return;
     }
+
+    throw Error{
```

Comment:
Don't get the idea of this change in if logic, the previous was more readable

Reply:
Reverted